Kinvey/Kinvey/CacheManager.swift

```diff
@@ -81,10 +81,13 @@ internal class CacheManager: NSObject {
                     if let encryptionKey = encryptionKey {
                         realmConfiguration.encryptionKey = encryptionKey
                     }
-                    if let realm = try? Realm(configuration: realmConfiguration) , !realm.isEmpty {
+                    if let realm = try? Realm(configuration: realmConfiguration), !realm.isEmpty {
                         try! realm.write {
                             realm.deleteAll()
                         }
+                        realm.invalidate()
```

Comment:
invalidate() will detach any existing live objects from this Realm instance
refresh() will reevaluate the Realm instance in case another Realm instance (maybe from another Thread) made any change in the database
and the last one is just to make sure that after all this the database is indeed empty
I added all these cleanups when I was trying to clean the state of the database in the end of each unit test (tearDown) and I was seeing some weird states. Not sure we if really need this. I think I will remove that.